### PR TITLE
Download visible activity layer for selected area // MAP-1219

### PR DESCRIPTION
**Pull request by fertandil87**
https://globalfishingwatch.atlassian.net/browse/MAP-1219

https://github.com/GlobalFishingWatch/frontend/assets/1672451/33b6a891-106c-427a-982c-271809b91d56

image

### CODE REVIEW

**Comment by fertandil87** [comment]
the test that fails is other that Natalia fixed in another PR

File: apps/fishing-map-e2e/src/e2e/map/eez-download.cy.ts [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+const verifyFileDownload = (file: string, button: string) => {
+  cy.intercept(API_URL_4WINGS_REPORT).as('downloadReport')
+  cy.getBySel(`download-${button}-button`).should('not.be.disabled').click()
+  // This request takes for me 17s, so I use 30 just in case
+  cy.wait('@downloadReport', { requestTimeout: 40000 })
+
+  //Create the same file name that the component CSVLink is generating
+  cy.readFile(file, { timeout: 15000 })
+}
```
Comment (rdgfuentes) [inline apps/fishing-map-e2e/src/e2e/map/eez-download.cy.ts:71]: Common functions used many times inside a page test could go in a PageObject file (eg: `apps/fishing-map-e2e/src/support/map/eez-download.po.ts`) and import them here.

Some repetitive actions inside tests (like [these](https://github.com/GlobalFishingWatch/frontend/pull/2154/files#diff-adc1e64389f81e8dcfdb8819c4468f695ab2ce022cfbb0c03fce50699dd6942bR147-R157))  could also be refactored and extracted to functions inside the `eez-download.po.ts` and invoked here.

File: apps/fishing-map-e2e/src/e2e/map/eez-download.cy.ts [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+    cy.log('DOWNLOAD JSON - GROUP BY GEAR AND DAY')
+    deleteDownloadsFolder()
+    selectOptions(REPORT_FORMAT_JSON, GROUPBY_GEAR, GROUPBY_DAY)
+    verifyFileDownload(jsonPath, 'activity-vessel')
+    verifyJson(['date', 'geartype', 'hours', 'vesselIDs'])
+  })
+
+  it('Should download gridded activity for ' + SEARCH_EEZ, () => {
+    deleteDownloadsFolder()
```
Comment (rdgfuentes) [inline apps/fishing-map-e2e/src/e2e/map/eez-download.cy.ts:161]: Why deleting the downloadsFolder as first step of the test if it is already deleted [beforeEach](https://github.com/GlobalFishingWatch/frontend/pull/2154/files#diff-adc1e64389f81e8dcfdb8819c4468f695ab2ce022cfbb0c03fce50699dd6942bR78) test?

File: apps/fishing-map-e2e/src/constants/search.ts [hunk 3]
```diff
@@ -3,3 +3,21 @@ export const SEARCH_VESSEL_MMSI = '259539000'
...
+  'Vessel Type',
+  'MMSI',
+  'IMO',
+  'CallSign',
+  'First Transmission Date',
+  'Last Transmission Date',
+  'Apparent Fishing Hours',
+]
+
+export const REPORT_CSV_GROUPED_COLUMNS = ['Time Range', 'Vessel IDs', 'Apparent Fishing Hours']
```
Comment (rdgfuentes) [inline apps/fishing-map-e2e/src/constants/search.ts:23]: These constants are never used, right? If so remove them please.

File: apps/fishing-map-e2e/src/constants/buttons.ts [hunk 4]
```diff
@@ -1,3 +1,14 @@
...
+export const REPORT_FORMAT_JSON = 'report-format-json'
+export const REPORT_FORMAT_CSV = 'report-format-csv'
+
+export const GROUPBY_FLAG = 'group-vessels-by-flag'
+export const GROUPBY_MMSI = 'group-vessels-by-mmsi'
+export const GROUPBY_GEAR = 'group-vessels-by-gearType'
+export const GROUPBY_FLAG_GEAR = 'group-vessels-by-flagAndGearType'
+
+export const GROUPBY_DAY = 'group-time-by-daily'
+export const GROUPBY_MONTH = 'group-time-by-monthly'
```
Comment (rdgfuentes) [inline apps/fishing-map-e2e/src/constants/buttons.ts:14]: If these constants are exclusively used in the `eez-download` test, I'd better move them to `eez-download.po.ts` as explained [below](https://github.com/GlobalFishingWatch/frontend/pull/2154/files#diff-adc1e64389f81e8dcfdb8819c4468f695ab2ce022cfbb0c03fce50699dd6942bR71)

**Review by rdgfuentes — CHANGES_REQUESTED** [review]

File: apps/fishing-map-e2e/src/support/commands.ts [hunk 5]
```diff
@@ -165,6 +169,24 @@ Cypress.Commands.add(
...
+  { prevSubject: true },
+  (
+    subject: Cypress.Chainable<HTMLElement>,
+    selector: string,
+    options?: Partial<Cypress.Loggable & Cypress.Timeoutable & Cypress.Withinable & Cypress.Shadow>
+  ) => {
+    Cypress.log({
+      name: 'findBySel',
+      displayName: 'FIND BY DATA-TEST',
+      message: [`💸 Try to find elements by data-test that starts with ${selector} in:`],
```
[the remark this answers is not in the document]
Comment (fertandil87) [reply apps/fishing-map-e2e/src/support/commands.ts:183]: its different, its not a like pattern

File: apps/fishing-map-e2e/src/constants/buttons.ts (hunk 4 above)
Comment (fertandil87) [reply apps/fishing-map-e2e/src/constants/buttons.ts:14]: I dont now if this will be used only for this test, so I think they can stay here

**Review by rdgfuentes — APPROVED** [review]

**Pull request merged**